sdks/java/core/src/main/java/org/apache/beam/sdk/options/PipelineOptionsFactory.java

```diff
@@ -1577,7 +1589,7 @@ public int compare(Method o1, Method o2) {
    * the expected java type using an {@link ObjectMapper} will be ignored.
    */
   private static <T extends PipelineOptions> Map<String, Object> parseObjects(
-      Class<T> klass, ListMultimap<String, String> options, boolean strictParsing) {
+      Class<T> klass, Multimap<String, String> options, boolean strictParsing) {
```

Comment:
Its important that this is a listmultimap and not a setmultimap as the command line "--foo=a --foo=a" should be equivalent to "--foo=a,a". Also, the parameters on the command line should not be re-ordered.

Reply:
If that's the spec, then `List<Pair<String, List<String>>>` makes the spec in the type, which is a bit better.

Reply:
@lukecwik agree, multimap is just more convenient and since it is an internal it is fine. If you check org.apache.beam.sdk.options.PipelineOptionsFactory.Builder#as I really use a list flavor of the multimap to guarantee it and parseCommandLine does too. The map case doesnt care. So we are all good.
@kennknowles for this internal structure `List<Pair<String, List<String>>>` is harder to use cause you miss the put for the usage which is done of the structure, probably better to avoid useless loops in the manipulation, no?

Reply:
Multimap is to vague on the contract, please stick with ListMultimap since it clearly specifies it.

This also documented in the Javadoc for Multimap: "Instead of using the Multimap interface directly, prefer the subinterfaces ListMultimap and SetMultimap. These take their names from the fact that the collections they return from get behave like (and, of course, implement) List and Set, respectively."